api/src/main/java/org/pmiops/workbench/cdr/dao/CustomCBCriteriaDaoImpl.java

```diff
@@ -0,0 +1,149 @@
+    }
+
+    public String getQuery() {
+      return query;
+    }
+
+    public MapSqlParameterSource getParameters() {
+      return parameters;
+    }
+  }
+
+  private static final String ENDS_WITH_WITHOUT_TERM =
+      "select *\n"
+          + "from %s.cb_criteria\n"
+          + "where is_standard = :standard\n"
+          + "and match(full_text) against(concat('+[', :domain, '_rank1]') in boolean mode)\n"
+          + "and (%s)\n"
+          + "order by est_count desc, name asc\n";
+
+  private static final String ENDS_WITH_WITH_TERM =
+      "select *\n"
+          + "from %s.cb_criteria\n"
+          + "where is_standard = :standard\n"
+          + "and match(full_text) against(concat(:term, '+[', :domain, '_rank1]') in boolean mode)\n"
+          + "and (%s)\n"
+          + "order by est_count desc, name asc\n";
+
+  private static final String DYNAMIC_SQL = "upper(name) like upper(%s)";
+
+  private static final String LIMIT_OFFSET = "limit %s offset %s\n";
+
+  private static final String OR = "\nor\n";
+
+  @Autowired private NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+
+  @Autowired private CdrVersionDao cdrVersionDao;
+
+  @Override
+  public Page<DbCriteria> findCriteriaByDomainAndStandardAndNameEndsWith(
+      String domain, Boolean standard, List<String> endsWithList, Pageable page) {
+    QueryAndParameters queryAndParameters =
+        generateQueryAndParameters(ENDS_WITH_WITHOUT_TERM, domain, standard, null, endsWithList);
+    return new PageImpl<>(
+        queryForPaginatedList(page, queryAndParameters),
+        page,
+        Objects.requireNonNull(count(queryAndParameters)));
+  }
+
+  @Override
+  public Page<DbCriteria> findCriteriaByDomainAndStandardAndTermAndNameEndsWith(
+      String domain, Boolean standard, String term, List<String> endsWithList, Pageable page) {
+    QueryAndParameters queryAndParameters =
+        generateQueryAndParameters(ENDS_WITH_WITH_TERM, domain, standard, term, endsWithList);
+    return new PageImpl<>(
+        queryForPaginatedList(page, queryAndParameters),
+        page,
+        Objects.requireNonNull(count(queryAndParameters)));
+  }
+
+  protected QueryAndParameters generateQueryAndParameters(
+      String sql, String domain, Boolean standard, String term, List<String> endsWithList) {
+    MapSqlParameterSource parameters = new MapSqlParameterSource();
+    parameters.addValue("domain", domain).addValue("standard", standard);
+    if (term != null) {
+      parameters.addValue("term", term);
+    }
+    StringJoiner joiner = new StringJoiner(OR);
+
+    long cdrVersionId = CdrVersionContext.getCdrVersion().getCdrVersionId();
+    Optional<DbCdrVersion> cdrVersionOptional = cdrVersionDao.findById(cdrVersionId);
+    DbCdrVersion dbCdrVersion =
+        cdrVersionOptional.orElseThrow(
+            () ->
+                new BadRequestException(
+                    String.format("CDR version with ID %s not found", cdrVersionId)));
```

Comment:
Lookup up the cdr version in the database so we can get the specific name: EX: r_2022q2_4